### DISCUSSION TITLE
Obstruction detection

### DISCUSSION THREAD

**Discussion by Sceptersax** [blakeblackshear/frigate#7090] [Ideas]
Hello. I’m new to the community. I’ve searched for the feature of detecting camera obstruction but I haven’t seen any posts about it. Maybe I missed it but I find it kind of odd considering the rise in crime in recent times. Having the ability to detect when the camera is blocked or obstructed would seem like a necessary analytic. 

The use case would be if someone were to be behind a camera and they place tape or other such obstruction over the camera lens and thereby circumvent any detection. There would be no human presence detected so in that scenario, we wouldn’t be able receive any alerts. And even if motion detection without human detection is possible (as far as I know it’s not in frigate), the case could be made that you would still dismiss it as another false positive. I’ve come across many cameras where this feature is actually built in the camera.

Please let me know if this is already a thing or on the roadmap. Thanks!

**Comment by NickM-27** [comment]
Frigate does support motion detection, both internally and exposes sensors in MQTT / websocket. 

Also, this is covered by a pinned feature request https://github.com/blakeblackshear/frigate/issues/1793

**Comment by Sceptersax** [comment]
Thanks @NickM-27. So I guess the solution would be to use HA to detect when the IR has been triggered? Not really a solution for me but I guess it’s a start. I’ve been really hesitant to start using HA because I feel like it would be a bit of a learning curve and have no idea how to implement this. If anyone can give me some advice or even an explanation on how to achieve this, I’d be really grateful. And I wish I was better versed on setting up MQTT websockets. I suppose I will google that.
